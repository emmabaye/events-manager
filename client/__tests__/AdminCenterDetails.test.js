import ConnectedAdminCenterDetails, { CenterDetails } from '../components/AdminCenterDetails.jsx';

describe('AdminCenterDetails Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      center: {
        data: [{
          id: '1',
          name: 'City Hall',
          description: {
            substr: () => {}
          },
          location: {
            substr: () => {}
          },
          capacity: "500",
          price: "50000",
          image: "",
          available: 'false',
          Events: []
        }]
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    centerId: "1",
    center: initialState.centerReducer.center
  };
  props.center.Events = [];

  describe('AdminCenterDetails form  should render self', () => {
    it('it should render for dumb component for center not available', () => {
      const wrapper = shallow(<CenterDetails {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render for dumb component for center available', () => {
      props.center.available = 'true';
      const wrapper = shallow(<CenterDetails {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedAdminCenterDetails /></Provider>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
