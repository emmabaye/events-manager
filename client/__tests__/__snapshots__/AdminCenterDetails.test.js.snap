// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminCenterDetails Component AdminCenterDetails form  should render self it should render for dumb component for center available 1`] = `
<div
  className=" panel center-details"
  id="center-details"
>
  <div
    className="container event-details"
  >
    <div
      className="row"
    >
      <div
        className=" event"
      >
        <div
          className="card mb-3"
        >
          <img
            alt="Card image cap"
            className="card-img-top img-fluid"
            height="300px"
          />
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b />
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <b>
                Capacity: 
                 seats
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Facilities: 
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Address: 
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Price: ₦
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Availability: 
                Center is Available
              </b>
            </p>
          </div>
        </div>
        <div
          className="row venue-date"
        >
          <div
            className="container"
          >
            <Location
              location="undefined, undefined"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AdminCenterDetails Component AdminCenterDetails form  should render self it should render for dumb component for center not available 1`] = `
<div
  className=" panel center-details"
  id="center-details"
>
  <div
    className="container event-details"
  >
    <div
      className="row"
    >
      <div
        className=" event"
      >
        <div
          className="card mb-3"
        >
          <img
            alt="Card image cap"
            className="card-img-top img-fluid"
            height="300px"
          />
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b />
            </h5>
            <p
              className="card-text"
            />
            <p
              className="card-text"
            >
              <b>
                Capacity: 
                 seats
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Facilities: 
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Address: 
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Price: ₦
              </b>
            </p>
            <p
              className="card-text"
            >
              <b>
                Availability: 
                Not Available
              </b>
            </p>
          </div>
        </div>
        <div
          className="row venue-date"
        >
          <div
            className="container"
          >
            <Location
              location="undefined, undefined"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
